### ISSUE TITLE
The object check on subscribe() is to restricting and doesn't allow functions

### ISSUE THREAD

**Issue by far-blue** [cujojs/most#445]
### Summary
While RsJS and zen-observable allow the object passed to subscribe() to be an object or a function, most.js specifically checks for typeof object which excludes functions.

### Expected result

subscribe(f) where f is a function object is accepted

### Actual Result

subscribe(f) where f is a function object throws an exception

### Versions

- most.js: 1.3

### Steps to reproduce

1. create a function with a next() method
1. pass it to subscribe()

### Code to reproduce
```
function f () {}
f.next = function (value) {}
most.from([1, 2, 4]).subscribe(f);
```

### Fix
Change the line in subscribe() from:
`if (subscriber == null || typeof subscriber !== 'object') {`

to the style of check zen-observable does:
`if (Object(subscriber) !== subscriber) {`

or just check for a function directly:
`if (subscriber == null || (typeof subscriber !== 'object' && typeof subscriber !== 'function')) {`

### Why allow functions?
I'm actually discussing with the knockoutjs team how they could keep backward compatibility of knockoutjs while providing interoperability with the TC39 Observable spec proposal. Knockout 'observables' are functions which we can extend to support the Observable and Observer specifications but only if an Observer is allowed to be a function. The spec. is not entirely clear but as RxJS and zen-observable appear to allow it and it offers a way for other libraries such as knockout to interop I wanted to request the change.

**Comment by far-blue** [comment]
The knockoutjs discussion is here: https://github.com/knockout/tko/issues/22

**Comment by briancavalier** [comment]
Hey @far-blue, thanks.  It does seem like both [zen-observable](https://github.com/zenparsing/zen-observable/blob/master/src/Observable.js#L85) and [rxjs allow it](https://github.com/ReactiveX/rxjs/blob/master/src/util/toSubscriber.ts), so it seems like most should line up with that.  Would you be up for sending a PR that switches to the `Object(subscriber)` check?

**Comment by far-blue** [comment]
There you go :)

**Comment by briancavalier** [comment]
Thanks @far-blue!  I'll release an update soon.

Fixed by #448
